compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -392,6 +469,12 @@ impl<'a, 'tcx> Visitor<'tcx> for LifetimeContext<'a, 'tcx> {
     // We want to nest trait/impl items in their parent, but nothing else.
     fn visit_nested_item(&mut self, _: hir::ItemId) {}
 
+    fn visit_trait_item_ref(&mut self, ii: &'tcx hir::TraitItemRef) {
+        if !self.definition_only {
```

Comment:
Do we ever *want* to do this? When we are resolving the trait, `definition_only` is true, and I think trait items only appear as the children of a trait, right?